### PR TITLE
fix/Locked but expired objects

### PR DESCRIPTION
**Pull request by carpawell**

### CODE REVIEW

**Comment by carpawell** [comment]
It waits https://github.com/nspcc-dev/neofs-node/pull/2526 but some changes will be also added.

**Review by roman-khimov — APPROVED** [review]

File: pkg/local_object_storage/engine/inhume.go [hunk 1]
```diff
@@ -179,7 +179,8 @@ func (e *StorageEngine) inhumeAddr(addr oid.Address, prm shard.InhumePrm, checkE
 	return ok, retErr
 }
 
-func (e *StorageEngine) isLocked(addr oid.Address) (bool, error) {
+// IsLocked checks whether an object is locked according to StorageEngine's state.
+func (e *StorageEngine) IsLocked(addr oid.Address) (bool, error) {
```
Comment (cthulhu-rider) [inline pkg/local_object_storage/engine/inhume.go:183]: previously was called under `execIfNotBlocked` condition only, pls double-check do we need to keep this or not

File: pkg/local_object_storage/shard/gc.go [hunk 2]
```diff
@@ -431,8 +431,17 @@ func (s *Shard) HandleExpiredLocks(lockers []oid.Address) {
...
+		s.log.Warn("expired object filtering",
+			zap.Error(err),
+		)
+
+		return
+	}
+
 	var pInhume meta.InhumePrm
-	pInhume.SetAddresses(lockers...)
+	pInhume.SetAddresses(append(lockers, expired...)...)
```
Comment (cthulhu-rider) [inline pkg/local_object_storage/shard/gc.go:444]: lets reflect this in method docs? same for `HandleDeletedLocks`

File: pkg/local_object_storage/shard/gc.go [hunk 3]
```diff
@@ -442,7 +442,7 @@ func (s *Shard) HandleExpiredLocks(lockers []oid.Address) {
 
 	var pInhume meta.InhumePrm
 	pInhume.SetAddresses(append(lockers, expired...)...)
-	pInhume.SetGCMark()
+	pInhume.SetForceGCMark()
```
[the remark this answers is not in the document]
Comment (carpawell) [reply pkg/local_object_storage/shard/gc.go:445]: > lockers  are unlocked cuz they cannot be locked according to the system protocol

yes, but they can not be removed regularly. that was the last [decision we made](https://github.com/nspcc-dev/neofs-node/issues/1227#issuecomment-1083186577) (and that is true for the current implementation)

> finally, could u pls clarify this to me again?

yeap, lockers are the only reason for that force GC mark

> and btw u didn't add similar change to HandleDeletedLocks

... and missing lockers above in the removing list is the reason not to use force GC

File: pkg/local_object_storage/shard/gc.go (hunk 2 above)
Comment (carpawell) [reply pkg/local_object_storage/shard/gc.go:444]: done

File: pkg/local_object_storage/engine/inhume.go (hunk 1 above)
Comment (carpawell) [reply pkg/local_object_storage/engine/inhume.go:183]: fixed

**Review by cthulhu-rider — APPROVED** [review]

**Pull request merged**